File: ReactiveRest/src/main/java/a/gleb/reactiverest/models/Account.java
package a.gleb.reactiverest.models;

import lombok.Data;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Set;

@Data
public class Account implements UserDetails {

    private Long id;
    private String username;
    private String password;
    private String name;
    private String surname;
    private String email;
    private boolean active;
    private Set<AccountRole> roleDTO;
    private Collection<PostModel> postDTOList = new ArrayList<>();


    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return List.of(new SimpleGrantedAuthority(roleDTO.toString()));
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: web_client/lib/models/account.dart
class Account {
  final int id;
  final String username;
  final String password;
  final String name;
  final String surname;
  final String email;
  /*final bool active;
  final Role role;*/

  Account({
    required this.id,
    required this.username,
    required this.password,
    required this.name,
    required this.surname,
    required this.email,
    /*required this.active,
    required this.role,*/
  });

  factory Account.fromJson(Map<String, dynamic> json){
    return Account(
        id: json['id'],
        username: json['username'],
        password: json['password'],
        name: json['name'],
        surname: json['surname'],
        email: json['email']
      /* active: json['active'],
        role: json['role']*/
    );
  }

}

enum Role{
  administrator,
  user
}